tests/test_testing_fixtures.py

```diff
@@ -0,0 +1,162 @@
+    
+    token = create_test_token(
+        role="admin",
+        additional_claims=additional_claims
+    )
+    
+    # Verify token was created
+    assert isinstance(token, str)
+    assert len(token.split('.')) == 3  # JWT has 3 parts
+
+
+def test_mock_openid_fixture(mock_openid):
+    """Test that the mock_openid fixture works correctly"""
+    # Verify the fixture returns a respx mock
+    assert hasattr(mock_openid, 'get')
+    
+    # Test that we can make a request to the mocked endpoint
+    import httpx
+    response = httpx.get(openid_config_url())
+    assert response.status_code == 200
+    assert response.json() == openid_configuration()
+
+
+@pytest.mark.asyncio
+async def test_mock_zitadel_auth_call_with_request_state():
+    """Test the MockZitadelAuth __call__ method with request state handling"""
+    from unittest.mock import Mock
+    
+    mock_auth = MockZitadelAuth(
+        issuer_url=ZITADEL_ISSUER,
+        app_client_id=ZITADEL_CLIENT_ID,
+        project_id=ZITADEL_PROJECT_ID,
+        allowed_scopes={"openid": "OpenID access", "profile": "Profile access"},
+        mock_user_id="test-user-456",
+        mock_scopes=["openid", "profile"]
+    )
+    
+    # Create a mock request object with state
+    mock_request = Mock()
+    mock_request.state = Mock()
+    
+    # Create mock security scopes
+    mock_security_scopes = Mock()
+    mock_security_scopes.scopes = ["openid"]
+    
+    # Call the mock auth
+    user = await mock_auth(mock_request, mock_security_scopes)
+    
+    # Verify the user was created correctly
+    assert user.claims.sub == "test-user-456"
+    assert user.claims.iss == ZITADEL_ISSUER
+    assert user.claims.client_id == ZITADEL_CLIENT_ID
+    assert user.access_token == "mock-access-token"
+    
+    # Verify the user was set in request state
+    assert mock_request.state.user == user
+
+
+@pytest.mark.asyncio
+async def test_mock_zitadel_auth_call_without_request_state():
+    """Test the MockZitadelAuth __call__ method without request state"""
+    from unittest.mock import Mock
+    
+    mock_auth = MockZitadelAuth(
+        issuer_url=ZITADEL_ISSUER,
+        app_client_id=ZITADEL_CLIENT_ID,
+        project_id=ZITADEL_PROJECT_ID,
+        allowed_scopes={"openid": "OpenID access"},
+        mock_user_id="test-user-789",
+        mock_scopes=["openid"]
+    )
+    
+    # Create a mock request object without state attribute
+    mock_request = Mock(spec=[])  # Empty spec means no attributes
+    
+    # Create mock security scopes
+    mock_security_scopes = Mock()
+    
+    # Call the mock auth
+    user = await mock_auth(mock_request, mock_security_scopes)
+    
+    # Verify the user was created correctly
+    assert user.claims.sub == "test-user-789"
+    assert user.claims.iss == ZITADEL_ISSUER
+    assert user.claims.client_id == ZITADEL_CLIENT_ID
+    assert user.access_token == "mock-access-token"
+    
+    # Verify no error occurred when request has no state attribute
```

Comment:
should a error rather be raised?

Reply:
Removed this comment; as if an error is raised it will be raised anyways.